heroku3/models/configvars.py

```diff
@@ -1,82 +1,108 @@
-import json
+import copy
 
 
 class ConfigVars(object):
-    """Heroku ConfigVars."""
+    """Set of configuration variables of a Heroku application
 
-    def __init__(self):
+    This class mutates the settings of the application as you mutate it's
+    content:
 
-        self.data = {}
-        self.app = None
-        self._h = None
+    .. code-block::
+
+       configuration = app.config()  # Assume app is an application
+       configure['SOME_SETTING'] = 'some value'
+
+    That last line of code will synchronously trigger a call to Heroku's
+    API to change the setting value. If this is not what you desire, you
+    can collect settings in a regular dictionary with::
+
+       import copy
+       configuration = app.config()  # Assume app is an application
+       safe_configuration = configuration.dict()
+       # mutate safe_configuration as you please
+       configuration.update(safe_configuration)
+
+    """
+
+    def __init__(self, data, app, h=None, **kwargs):
+        # assert app is not None
+        # copy: Ensures caller has no way to mutate our internal state outside
+        # our control
+        self.__data = copy.copy(data)
+        self.__app = app
+        self._h = h
 
         super(ConfigVars, self).__init__()
 
     def __repr__(self):
-        return repr(self.data)
+        return repr(self.__data)
 
     def __contains__(self, key):
-        v = None
-        try:
-            v = self[key]
-        except KeyError:
-            return False
-        else:
-            if v is None:
-                return False
-            else:
-                return True
+        return key in self.__data
```

Comment:
Much simpler. Assumes local object state is in sync with the remote state of the configuration though.